leshan-client-core/src/main/java/org/eclipse/leshan/client/servers/ServersInfoExtractor.java

```diff
@@ -102,9 +102,13 @@ public static ServersInfo getInfo(Map<Integer, LwM2mObjectEnabler> objectEnabler
 
                         boolean useOscore = oscoreObjLink.getObjectId() == OSCORE;
```

Comment:
oscoreObjLInk can probably be null as OSCORE SECURITY MODE is optional.

So Regarding issue above maybe we can change the code like this ?  
```java 
// find associated oscore instance (if any)
LwM2mObjectInstance oscoreInstance = null;
ObjectLink oscoreObjLink = (ObjectLink) security.getResource(SEC_OSCORE_SECURITY_MODE)
        .getValue();
if (oscoreObjLink != null) {
    int oscoreObjectInstanceId = oscoreObjLink.getObjectInstanceId();
    if (!oscoreObjLink.isNullLink() && oscoreObjLink.getObjectId() != OSCORE) {
        LOG.warn(
                "The security object's 'OSCORE Security Mode' links to an incorrect object type.");
    } else {
        oscoreInstance = oscores.getInstance(oscoreObjectInstanceId);
        if (oscoreInstance == null) {
            // maybe we should use same log level for this one and the other just above
            LOG.error("Failed to retrieve OSCORE object linked from BS security object");
        }
    }
}

if (oscoreInstance != null) {
    info.useOscore = true;
    info.masterSecret = getMasterSecret(oscoreInstance);
    info.senderId = getSenderId(oscoreInstance);
    info.recipientId = getRecipientId(oscoreInstance);
    info.aeadAlgorithm = getAeadAlgorithm(oscoreInstance);
    info.hkdfAlgorithm = getHkdfAlgorithm(oscoreInstance);
    info.masterSalt = getMasterSalt(oscoreInstance);
```

(and same for register)